### DISCUSSION TITLE
Make available sales channel as Functions input (Online Store, POS, etc.)

### DISCUSSION THREAD

**Discussion by mdriess** [Shopify/function-examples#395] [Feedback: General Improvements]
Right now Functions seem to apply at POS but documentation isn't clear on this at all (every merchant we speak to that is sharp enough to know what Functions are is surprised that this is the case as it's not clear and not how Scripts worked); also Functions don't have a way to read if the order is POS or Online.

We definitely love Functions for POS, but we would like to as soon as possible be able to know the channel.

You can technically work around this kinda, but in bad ways:
- detect presence of shipping address (not a true test since ship to customer from POS is not really an online channel order)
- have online store add a line item attribute which POS won't have (obviously an annoying and messy option at best)

It would be amazing just to know the channel. POS location as an additional input would also be huge; I see that mentioned in #270

**Comment by pablogiralt** [comment]
Similar to this https://github.com/Shopify/function-examples/discussions/270

**Reply by mdriess** [reply]
I upvoted there as well :) I think here the goal is to consider all possible sales channels both current in future, not just POS vs Online. Agreed that POS locations would be amazing to be exposed.

**Comment by nickwesselman** [comment]
Moving to General as it applies to multiple APIs.

**Comment by overallduka** [comment]
Up on this, POS now has `retailLocation` attribute [https://shopify.dev/changelog/pos-specific-shopify-function-logic](url) on the function to identify, but Shop App remains without some attribute to identify the source of the traffic.
